### ISSUE TITLE
Hook doesn't work with `exclude` and `include`

### ISSUE THREAD

**Issue by paveldedik** [lk16/detect-missing-init#2]
Hi,
I have the following settings in pre-commit-config.yaml:

```yaml
    -   repo: https://github.com/lk16/detect-missing-init
        rev: v0.1.4
        hooks:
        -   id: detect-missing-init
            files: '^folder/.*'
            args: ['--create']

```

But the `files` seems to be ignored. I was also trying to use `exclude` but that doesn't work either.

Maybe the hook works differently and this is not possible, it would be nice to either:

- fix this behavior so that `exclude` and `files` works
- add configuration arguments which would allow similar behavior

**Comment by lk16** [comment]
Hi,
The `files` option is ignored because this hook [uses](https://github.com/lk16/detect-missing-init/blob/master/.pre-commit-hooks.yaml#L7) `pass_filenames = false` in the hook config. 

The reason for that is that we don't really check content of files (like most pre-commit hooks), but rather presence of files. Passing filenames doesn't work well, because there would be a lot of redundant work. 

Back to your feature request: If I understand correctly you want to be able to either whitelist or blacklist folders that need to be checked for a missing `__init__.py` file.

Currently there is already the `--skip-folders` flag, which does blacklisting, however this currently doesn't support patterns / globs.
Whitelisting is not implemented at all, except for the repo root, by using `--expect-root-init`. Here we could also add patterns / globs using some new flag like `--check-folders` or similar.

Could you confirm that I understood your requested features correctly? I think I can implement these things, unless you prefer to implement them yourself. PR's are welcome!

Thanks for checking out my project.

**Comment by paveldedik** [comment]
Hi, thank you for your answer.
I understand that this hook works a bit differently.

It would suffice to have opposite of `--skip-folders`, i.e. `--folders` or `--include-folders` option.

But as I was testing it, even the `--skip-folders` option doesn't seem to be working as expected, I have the following:

```yaml
    -   repo: https://github.com/lk16/detect-missing-init
        rev: v0.1.4
        hooks:
        -   id: detect-missing-init
            args: ['--skip-folders=client']
```

The client folder is in the root (where `.pre-commit-config.yaml` is) and it is NOT being skipped.

And BTW yes, I think you understand what I need :-) Right now, I would be ok to just have an option to include a nested folder in my repository with an option like `--folders=src/app` where `__init__.py` files would be created, all other folders outside would be ignored.

**Comment by lk16** [comment]
> Right now, I would be ok to just have an option to include a nested folder in my repository with an option like --folders=src/app where __init__.py files would be created, all other folders outside would be ignored.

I have just released version [v0.1.5](https://github.com/lk16/detect-missing-init/releases/tag/v0.1.5), which introduces the `--python-folders` flag, which does exactly that. Note that some older flags have been removed, to reduce clutter and prevent duplicated functionality.

**Comment by paveldedik** [comment]
Thank you very much! It works and resolves my issue. Feel free to close the issue.
